src/file.ts

```diff
@@ -3644,20 +3643,6 @@ class File extends ServiceObject<File, FileMetadata> {
     }
     const returnValue = retry(
       async (bail: (err: Error) => void) => {
-        if (data instanceof Readable) {
```

Comment:
Removing this logic as for the vast majority it would simply slow down their uploads unnecessarily. Additionally, later versions of Node.js catch this for us.

Reply:
Do you know what versions catch this?

Reply:
Node 18.0.0:
- https://github.com/nodejs/node/pull/46600
- https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V18.md#18.0.0

Here's a simple reproduction for a REPL (anything before 18 the `console.log` isn't called):
```
const readable = stream.Readable({read() {this.push(null)}});
readable.read();
readable.readable // false
stream.pipeline(readable, stream.PassThrough, () => console.log('done'));
```